packages/client/src/rpc/validation.ts

```diff
@@ -303,6 +303,22 @@ export const validators = {
     }
   },
 
+  /**
+   * number validator to check if type is integer
+   * @param params parameters of method
+   * @param index index of parameter
+   */
+  get integer() {
```

Comment:
Strictly, an **integer** is:
> A member of the set of positive whole numbers {1, 2, 3, … }, negative whole numbers {−1, −2, −3, … }, and zero {0}. 

`Number.isInteger(n: unknown)` will return **false** *unless* `typeof n === 'number'` *and* **n** is an **integer** 

```suggestion
  
  get integer() {
    return (params: any[], index: number) => {
      if (!Number.isInteger(params[index])) {
        return {
          code: INVALID_PARAMS,
          message: `invalid argument ${index}: argument is not a number`,
        }
      }
    }
  },
```